src/Raven.Server/Documents/Indexes/Static/Spatial/SpatialDistanceFieldComparatorSource.cs

```diff
@@ -96,17 +142,85 @@ private double CalculateDistance(int doc, IState state)
                 }
                 catch (InvalidOperationException)
                 {
-                    return double.NaN;
+                    return SpatialResult.Invalid;
                 }
                 var pt = shape as Point;
                 if (pt == null)
                     pt = shape.GetCenter();
-                return _spatialField.GetContext().GetDistCalc().Distance(pt, _originPt);
+
+                var result = new SpatialResult
+                {
+                    Latitude = pt.GetY(),
+                    Longitude = pt.GetX()
+                };
+
+                if (_isGeo == false)
+                {
+                    result.Distance = CartesianDistance(pt.GetY(), pt.GetX(), _originPt.GetY(), _originPt.GetX());
+                    return result;
```

Comment:
what if we want KM?

Reply:
That isn't how it works for cartesian, that is arbitrary system